app/src/main/java/com/google/android/samples/socialite/ui/ContactRow.kt

```diff
@@ -40,60 +45,81 @@ import androidx.compose.ui.unit.sp
 import com.google.android.samples.socialite.data.utils.toReadableString
 import com.google.android.samples.socialite.model.ChatDetail
 
+@OptIn(ExperimentalSharedTransitionApi::class)
 @Composable
 fun ChatRow(
     chat: ChatDetail,
     onClick: (() -> Unit)?,
     modifier: Modifier = Modifier,
+    sharedTransitionScope: SharedTransitionScope,
+    animatedContentScope: AnimatedContentScope
 ) {
-    Row(
-        modifier = modifier
-            .fillMaxWidth()
-            .then(
-                if (onClick != null) Modifier.clickable(onClick = onClick) else Modifier,
-            )
-            .padding(16.dp),
-        verticalAlignment = Alignment.CenterVertically,
-        horizontalArrangement = Arrangement.spacedBy(16.dp),
-    ) {
-        // This only supports DM for now.
-        val contact = chat.attendees.first()
-        Image(
-            painter = rememberIconPainter(contentUri = contact.iconUri),
-            contentDescription = null,
-            modifier = Modifier
-                .size(48.dp)
-                .clip(CircleShape)
-                .background(Color.LightGray),
-        )
-        Column(
-            modifier = Modifier.weight(1f),
-            verticalArrangement = Arrangement.SpaceEvenly,
-            horizontalAlignment = Alignment.Start,
-        ) {
-            Text(
-                text = contact.name,
-                style = MaterialTheme.typography.bodyLarge,
-                fontSize = 16.sp,
-            )
-            Text(
-                text = chat.chatWithLastMessage.text,
-                style = MaterialTheme.typography.bodySmall,
-                maxLines = 1,
-                overflow = TextOverflow.Ellipsis,
-                modifier = Modifier.padding(top = 4.dp),
-                fontWeight = FontWeight.Light,
-                fontSize = 14.sp,
-            )
-        }
-        Column(
-            verticalArrangement = Arrangement.SpaceEvenly,
-            horizontalAlignment = Alignment.CenterHorizontally,
+    with(sharedTransitionScope) {
+        Row(
+            modifier = modifier
+                .fillMaxWidth()
+                .then(
+                    if (onClick != null) Modifier.clickable(onClick = onClick) else Modifier,
+                )
+                .padding(16.dp)
+                .sharedBounds(
+                    sharedTransitionScope.rememberSharedContentState(key = "Bounds${chat.chatWithLastMessage.id}"),
```

Comment:
If this shared element is quite complex - you might want to consider introducing a class that represents the key - For example like this https://developer.android.com/develop/ui/compose/animation/shared-elements#unique-keys. It means it'll be strongly typed, easier to navigate the codebase and find the corresponding keys, and harder to override if you were to add another shared element on the same screen.